tensorflow_probability/python/bijectors/real_nvp.py

```diff
@@ -266,8 +266,11 @@ def real_nvp_default_template(hidden_layers,
 
   with tf.name_scope(name, "real_nvp_default_template"):
 
-    def _fn(x, output_units):
+    def _fn(x, output_units, **condition_kwargs):
       """Fully connected MLP parameterized via `real_nvp_template`."""
```

Comment:
Actually I have a quick question here:

I'm thinking that the default should actually do no conditioning. It's not clear to me that doing tf.concat([x] + [condition_kwargs]] is the right thing to do / pass in.  I.e:  I can imagine other equally sane ways to have a MLP conditioned on this. I might suggest reverting the change here. WDYT?

Reply:
I agree that there might be other ways users might want to condition their networks. I thought providing some sane default here would be nice. After all, this still remains usable in the default form when the `condition_kwargs` is empty.

Happy to revert this if you think it's better to keep the default clean from the conditions.

Reply:
That's fair. Yeah I imagined it looked a little cleaner with out it, but I'm fine with the change as is. I'll start the process for pulling this in for an internal review, and see if anyone has any concerns with this (what will happen is this will go through an internal code review, and I will post back / notify you of changes /comments to make). Thanks again for this change!

Reply:
From talking internally, it seems like I think we might prefer to actually throw an exception for conditioning. concat'ing makes a fairly strong assumption that the condition_kwargs are all Tensors / Tensor-like. This might break when a user passes in a boolean condition_kwarg for instance. Also, in this case if they are tensors, then the user could also concat this to x and pass it in.

Can you instead raise NotImplementedError('Conditioning not implemented in default template.') or something of that nature?

Then in the test you can implement some version of this, and check if it works.

Thanks again!

Reply:
Makes a lot of sense, I now do I agree that maybe having this be the default is not the best option.

One thing we could do is have a `condition_fn` argument for the template, which the user could pass to customize the condition handling. Something like this:

```python
def real_nvp_default_template(hidden_layers,
                              shift_only=False,
                              activation=tf.nn.relu,
                              condition_fn=None,
                              name=None,
                              *args,  # pylint: disable=keyword-arg-before-vararg
                              **kwargs):
  with tf.name_scope(name, "real_nvp_default_template"):

    def _fn(x, output_units, **condition_kwargs):
      """Fully connected MLP parameterized via `real_nvp_template`."""
      if condition_kwargs and condition_fn is None:
        raise NotImplementedError(
            "Conditioning not implemented in the default template.")

      if condition_fn is not None:
        x = condition_fn(x, condition_kwargs)

      ...
```

And then the usage would be something like:
```python
def my_condition_fn(x, condition_kwargs):
    x = tf.concat(
        [x] + [condition_kwargs[k] for k in sorted(condition_kwargs)],
        axis=-1)
    return x

template = real_nvp_default_template(
    hidden_layers=[3],
    shift_only=False,
    condition_fn=my_condition_fn)
```

Just a thought. It would provide an easy way and an example of using condition. On the other hand, it might actually be too complicated considering the simplicity of the template function.

Reply:
Yeah I think preferring simplicity here would be what we want. I can imagine also more complicated forms of conditioning (e.g. the conditions come up in the network construction, and make more sense when building the layers, then when doing input preprocessing) could come up, and I guess this could confuse those sorts of users.

Perhaps the right thing to do here is to raise the NotImplementedError, but add in a docstring a snippet of how to do conditioning? What do you think?

Reply:
Yeah, you're totally right. I removed conditioning from the default template and added a small note to the docstring. Also changed tests accordingly.